### PR TITLE
rhc-7979: Restart POD when DB is not accessible by inventory-mq

### PR DESCRIPTION
**Pull request by thearifismail**
## Overview

This PR is being created to address [this Jira](https://issues.redhat.com/browse/RHCLOUD-7979).  As host-inventory consumes messages from Kafka and updates the DB and produce events, the `inventory-mq` containers may be find RDS inaccessible for any reason.  When that happens, the mq container should exit and the pod be restarted. 

## Secure Coding Practices Checklist GitHub Link

- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: tests/test_mq_service.py [hunk 1]
```diff
@@ -1062,3 +1064,44 @@ def test_host_account_using_mq(mq_create_or_update_host, api_get, db_get_host, d
 
     assert created_host.__dict__ == same_host.__dict__
     assert len(first_batch.all()) == len(second_batch.all())
+
+
+def test_handle_message_with_test_identity(mocker, flask_app):
+    mocker.patch("app.queue.queue.build_event")
+    fake_add_host = mocker.patch("app.queue.queue.add_host", return_value=(mocker.MagicMock(), None, None, None))
```
Comment (RonnyPfannschmidt) [inline tests/test_mq_service.py:1071]: you need to mock host_repository.add_host here

File: app/queue/queue.py [hunk 2]
```diff
@@ -166,6 +172,13 @@ def event_loop(consumer, flask_app, event_producer, handler, interrupt):
...
+                    except OperationalError as oe:
+                        """ sqlalchemy.exc.OperationalError: This error occurs when an
+                            authentication failure occurs or the DB is not accessible.
+                            Exit the process to restart the pod
+                        """
+                        logger.error(f"Could not access DB {str(oe)}")
+                        sys.exit(3)
                     except Exception:
                         metrics.ingress_message_handler_failure.inc()
                         logger.exception("Unable to process message")
```
Comment (tahmidefaz) [inline app/queue/queue.py:184]: Should we also increment some metrics for `OperationalError` as we do for other `Exceptions`?

**Review by tahmidefaz — CHANGES_REQUESTED** [review]

File: tests/test_mq_service.py (hunk 1 above)
Comment (thearifismail) [reply tests/test_mq_service.py]: This was a dummy test, I created to start but has been deleted since then.

File: app/queue/queue.py (hunk 2 above)
Comment (kruai) [reply app/queue/queue.py:184]: We should definitely do that for visibility purposes ^

Comment (thearifismail) [reply app/queue/queue.py:184]: Done!

**Review by kruai — APPROVED** [review]
I think this looks good now!

**Review by tahmidefaz — APPROVED** [review]

**Pull request merged**